### PR TITLE
[T3A3][W13-A1]DewiTan

### PR DESCRIPTION
**Pull request by jessidew95**

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,12 +8,19 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Toa Payoh Lorong 8, #03-10, 123456";
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: good that you have updated the help in the program. additionally you need to update:
1. user guide
2. tests

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -8,12 +8,19 @@
...
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person address should be in this format [Block], [Street], [Unit], [Postal Code]";
+    public static final String ADDRESS_VALIDATION_REGEX = 
+    		"[\\d+][\\,][\\s?][\\w+][\\,][\\s?]#[\\d+]-[\\d+][\\,][\\s?][\\d+]"; //block, address, unit and postal code all separated by ", "
+    
 
     public final String value;
     private boolean isPrivate;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: why is value still retained? you will introduce duplicity and inconsistency due to that

File: src/seedu/addressbook/data/person/Block.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: Please include a header comment to indicate what this class/method does?

File: src/seedu/addressbook/data/person/Block.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+public class Block {
+	
+	public static final String EXAMPLE = "123";
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: formatting is off...

File: src/seedu/addressbook/data/person/Block.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        return test.matches(BLOCK_VALIDATION_REGEX);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: this is good; you have take care of all overridden methods. :+1:

**Comment by okkhoy** [comment]
@jessidew95  overall I like your attempt, it is quite extensive.
please take a look at the comments, ack and close this PR once you have noted the comments.

**Comment by jessidew95** [comment]
noted!

**Comment by okkhoy** [comment]
great! please close the PR if you haven't done yet.

**Pull request closed without merging**